text/php/0001-restructure.md

```diff
@@ -54,15 +54,28 @@ buildpacks[<sup>2</sup>](#note-2):
   executable whenever possible.
 
 * **composer**:
-  Installs [`composer`](https://getcomposer.org), a dependency manager for PHP
-  and makes it available on the `$PATH`
+  Detect: always
   * provides: `composer`
-  * requires: none
+  * requires: none (will require `composer` if `BP_COMPOSER_VERSION` is set)
+  Build: Installs [`composer`](https://getcomposer.org), a dependency manager for PHP
+    and makes it available on the `$PATH`
+
+* **composer-global**:
+  Detect: When `BP_COMPOSER_INSTALL_GLOBAL` is set
+  * provides: none
+  * requires:
+    * `php` at build
+    * `composer` at build
+  Build: Runs `composer global` to ensure that global PHP packages are available on the `$PATH`
+      for composer install scripts
```

Comment:
What's the main rationale for having this split out from the `composer` buildpack?

Reply:
`composer-global` installs packages that are only meant for composer scripts, not for the PHP application itself. 

Its output should be on the `$PATH` for `composer-install` in the build container but should not be on the `$PATH` in the launch container, so I felt splitting it out would accomplish that.

Reply:
Could we add it as a step in `composer-install`?

Reply:
I think other install buildpacks do something similar

Reply:
Ok, it can be done inside `composer-install`.